### PR TITLE
Spring boot practice add shoppingcart

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/14-create-cart_items-table.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                  name: quantity
+                  type: integer
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (MinoDentori) [inline src/main/resources/db/changelog/changes/14-create-cart_items-table.yaml]: Add empty line 
Screenshot 2024-01-03 at 09 20 55

File: src/main/resources/db/changelog/changes/13-create-shopping_carts-table.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                  constraints:
+                    nullable: false
+                    references: users(id)
+                    foreignKeyName: fk_shopping_carts_user_id
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (MinoDentori) [inline src/main/resources/db/changelog/changes/13-create-shopping_carts-table.yaml]: Add empty line

**Review by MinoDentori — APPROVED** [review]
GJ. Minor comment to fix.

File: src/main/java/mate/academy/bookshop/controller/ShoppingCartConroller.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            @PathVariable(name = "id") Long cartItemId,
+            @RequestBody @Valid UpdateQuantityBookRequestDto requestDto
+    ) {
+        User user = (User)authentication.getPrincipal();
+        return shoppingCartService
+                .updateQuantityByCartItemId(user.getId(), cartItemId, requestDto);
+    }
+
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
+    @DeleteMapping("/cart-items/{id}")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/ShoppingCartConroller.java:65]:
```suggestion
    @DeleteMapping("/cart-items/{id}")
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
```

File: src/main/java/mate/academy/bookshop/dto/shoppingcart/AddToCartRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.bookshop.dto.shoppingcart;
+
+import jakarta.validation.constraints.Min;
+import lombok.Data;
+
+@Data
+public class AddToCartRequestDto {
+    private Long bookId;
+    @Min(value = 1, message = "can't be less than 1")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/dto/shoppingcart/AddToCartRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/mate/academy/bookshop/model/CartItem.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "cart_items")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    private ShoppingCart shoppingCart;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/CartItem.java:30]:
```suggestion
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private ShoppingCart shoppingCart;
```

File: src/main/java/mate/academy/bookshop/repository/cartitem/CartItemRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.bookshop.repository.cartitem;
+
+import mate.academy.bookshop.model.CartItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+public interface CartItemRepository
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/repository/cartitem/CartItemRepository.java:9]:
```suggestion
@Repository
public interface CartItemRepository
```

File: src/main/resources/db/changelog/changes/06-insert-roles.yaml
Comment (d1sam) [inline src/main/resources/db/changelog/changes/06-insert-roles.yaml:1]: remove this unnecessary change

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but there are some changes should be done!

**Review by vovasalyha — APPROVED** [review]
